airbyte-integrations/connectors/source-stripe/unit_tests/test_create_customers.py

```diff
@@ -0,0 +1,333 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+from unittest import TestCase
+from unittest.mock import Mock
+
+import pytest
+from tools.create_customers import (
+    BankAccount,
+    Customer,
+    HttpMethod,
+    HttpRequest,
+    StripeAPI,
+    create_customer_and_bank_account,
+    is_customer_already_created,
+)
+
+_MOCK_RESPONSE = {"key": "value"}
+_BEARER_TOKEN = "secret"
+_AUTHENTICATION_HEADER = {"Authorization": "Bearer secret"}
+
+
+class StripeApiTestCase(TestCase):
+    def setUp(self) -> None:
+        self._requester = Mock()
+        self._requester.submit.return_value = _MOCK_RESPONSE
+        self._stripe_api = StripeAPI("secret", self._requester)
+
+    def test_prepare_create_customer_request(self):
+        record = {
+            "description": "A fake customer",
+            "email": "fake@airbyte.io",
+            "name": "First Last",
+            "balance": "900001",
+            "bank_account": {
+                "object": "bank_account",
+                "country": "US",
+                "currency": "usd",
+                "account_holder_type": "individual",
+                "routing_number": "110000000",
+                "account_number": "000123456789",
+            },
+        }
+
+        customer = Customer.parse_obj(record)
+
+        expected_request = HttpRequest(
+            method=HttpMethod.POST,
+            url="https://api.stripe.com/v1/customers",
+            headers=_AUTHENTICATION_HEADER,
+            body={
+                "description": "A fake customer",
+                "email": "fake@airbyte.io",
+                "name": "First Last",
+                "balance": "900001",
+            },
+        )
+
+        response = self._stripe_api.create_customer(customer)
+
+        self._requester.submit.assert_called_with(expected_request)
```

Comment:
Sidenote: I think we should do more of those validations. I really miss mockito from Java where you can do something like `given(mock.method(param1, param2)).thenReturn(x)` where if the mock receives anything that does not fit `param1/param2`, `x` will not be returned. We can achieve the same with side_effects but the syntax is really not so nice